src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DisplayMetadata.cs

```diff
@@ -40,10 +41,10 @@ public class DisplayMetadata
         public string DisplayFormatString { get; set; }
 
         /// <summary>
-        /// Gets or sets a display name for the model.
+        /// Gets or sets a delegate used for getting the display name for the model.
         /// See <see cref="ModelMetadata.DisplayName"/>
         /// </summary>
-        public string DisplayName { get; set; }
+        public Func<string, string> DisplayName { get; set; }
```

Comment:
@IDisposable and @harshgMSFT I understand this makes it possible to use the same lambda for all properties of a `Type`. But I agree with @rynowak that this is an unlikely implementation choice. `Description` and `DisplayName` are assigned in a property-specific context. If the caller wants to deter use of the property name, they can do so without the API forcing this.